### PR TITLE
Fix activate widget tab no keep active after triggered save

### LINKED ISSUE #30166: Activate Tab no longer works on legacy tabs/forms
### Preconditions (*)
1. Magento 2.4-develop
2. Legacy Form with Left Tabs
3. URL with param: active_tab/TAB_NAME

### Steps to reproduce (*)
1. Create a new record in a form with multiple left tabs
2. Select one, other than the first left tab
3. Save the form

### Expected result (*)
1. Users should be redirected back to the form with the previous selected left tab active

### Actual result (*)
1. Users are sent back to the form, and the first left tab is always selected

### Media (Not Working 2.4)
1. Video: https://imgur.com/a/pnb3Gh1 

### Media (Working 2.3.5)
1. Video: https://imgur.com/a/TINgiNT

### Other Info

This bug was introduced by these two commits:

https://github.com/magento/magento2/commit/a3d9095b1352eb585399e7cce850112471955607#diff-926633484bdda3384356b3f61121a4e1

On line 34, the `<li>` element now has an ID attribute. The problem is that this ID is now duplicated, because the `<a>` element inside this `<li>` already has the same ID. And this now leads the "mage/backend/tabs" widget to fail loading the proper element

https://github.com/magento/magento2/commit/82f5a3696c55e65387fc330749e7267dcc269cf1#diff-926633484bdda3384356b3f61121a4e1
Added the CSS class 'no-display',

---

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [x] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

### PR DESCRIPTION
**Pull request by mrtuvn**
### Description (*)

Clean html class tabs (eliminate duplicate id for list item and anchor) and make sure tab work correctly like before
After this patch tab current keep active after admin change config and save form. Reload page return correct previous tab state
Screenshot from 2021-02-16 21-41-44

Before after save form reload and set active for very first tab

### Fixed Issues (if relevant)

1. Fixes magento/magento2#30166

### Manual testing scenarios (*)

1. Login admin
2. Go to Stores
3. Select Attributes in dropdown main menu
4. Go Edit Product attribute
5. Choose Storefront Properties tab (last tab)
6. Change some options in right content tab
7. Click` save and continue edit` and watch results

Expected results
When we selected tab ex: "Storefront Properties" and do some change in form. Then save. After page reload tab "Storefront" should keep active state instead

Actually results
The first tab active instead previous tab

### Questions or comments

CC: @bvboas hope this fix correct as your wish
### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Functional Tests CE, Functional Tests EE

**Comment by mrtuvn** [comment]
@magento run Functional Tests CE

**Review by edenduong — CHANGES_REQUESTED** [review]
Hi, 

Could you please cover this change by MFTF test?

Thanks !

**Comment by mrtuvn** [comment]
Is there any existed test related with file changed or i need create new one

**Comment by mrtuvn** [comment]
@magento give me test instance

**Comment by mrtuvn** [comment]
@engcom-Charlie @AnnaAPak @engcom-Kilo Can you guys help with MFTF coverage
Area affect is attribute product edit page

**Comment by mrtuvn** [comment]
CC @kate-kyzyma can you help with mftf coverage such case ?
This area in backend

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Functional Tests B2B, Functional Tests EE, Static Tests

**Comment by mrtuvn** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Comment by mrtuvn** [comment]
@kate-kyzyma Hi can you help cover changes by mftf test

**Comment by mrtuvn** [comment]
added mftf cover changes
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Unit Tests, WebAPI Tests, Integration Tests, Functional Tests EE

File: app/code/Magento/Catalog/Test/Mftf/Test/AdminVerifyStorefrontPropertiesTabStillActiveAfterSaveContinueEditTest.xml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            <magentoCron groups="index" stepKey="reindexInvalidatedIndices"/>
+        </after>
+
+        <!-- Go to Stores > Product, click "Add New Attribute" -->
+        <actionGroup ref="AdminOpenProductAttributePageActionGroup" stepKey="openProductAttributePage"/>
+        <click selector="{{AdminProductAttributeGridSection.createNewAttributeBtn}}" stepKey="createNewAttribute"/>
+        <waitForPageLoad stepKey="waitForNewAttributeFormLoad"/>
+
+        <!-- Verify properties storefront tab visible -->
+        <seeElement selector="{{StorefrontPropertiesSection.StoreFrontPropertiesTab}}" stepKey="seeTabAttributePropertiesVisible" />
```
Comment (edenduong) [inline app/code/Magento/Catalog/Test/Mftf/Test/AdminVerifyStorefrontPropertiesTabStillActiveAfterSaveContinueEditTest.xml]: I think you should create new ActionGroup

File: app/code/Magento/Catalog/Test/Mftf/Test/AdminVerifyStorefrontPropertiesTabStillActiveAfterSaveContinueEditTest.xml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        <!-- <scrollToTopOfPage stepKey="scrollToTabs"/> -->
+
+        <!-- Click on save continue -->
+        <click selector="{{AttributePropertiesSection.SaveAndEdit}}" stepKey="clickSave"/>
+
+        <!-- wait page reload done -->
+        <waitForPageLoad stepKey="waitForPageProductAttributeLoad"/>
+
+        <!-- Verify tab still active -->
+        <seeElement selector="//li[@id='list_product_attribute_tabs_front'][contains(@class, 'admin__page-nav-item ui-state-default ui-corner-top ui-tabs-active ui-state-active')]" stepKey="seeTabAttributePropertiesActive" />
```
Comment (edenduong) [inline app/code/Magento/Catalog/Test/Mftf/Test/AdminVerifyStorefrontPropertiesTabStillActiveAfterSaveContinueEditTest.xml]: Create new ActionGroup for assert the result

**Review by edenduong — CHANGES_REQUESTED** [review]
Thank you for you contribution. Please see my comment.

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Functional Tests CE, Functional Tests B2B, Functional Tests EE, Sample Data Tests B2B,  Sample Data Tests CE, Sample Data Tests EE

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by edenduong** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Functional Tests CE, Functional Tests B2B, Functional Tests EE

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run WebAPI Tests, Functional Tests CE, Functional Tests B2B

File: app/code/Magento/Catalog/Test/Mftf/Test/AdminVerifyStorefrontPropertiesTabStillActiveAfterSaveContinueEditTest.xml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+            <actionGroup ref="AdminLogoutActionGroup" stepKey="logoutAsAdmin"/>
+            <!-- Reindex invalidated indices after product attribute has been created/deleted -->
+            <magentoCron groups="index" stepKey="reindexInvalidatedIndices"/>
+        </after>
+
+        <!-- Go to Stores > Product, click "Add New Attribute" -->
+        <actionGroup ref="AdminOpenProductAttributePageActionGroup" stepKey="openProductAttributePage"/>
+        <click selector="{{AdminProductAttributeGridSection.createNewAttributeBtn}}" stepKey="createNewAttribute"/>
+        <waitForPageLoad stepKey="waitForNewAttributeFormLoad"/>
```
Comment (edenduong) [inline app/code/Magento/Catalog/Test/Mftf/Test/AdminVerifyStorefrontPropertiesTabStillActiveAfterSaveContinueEditTest.xml]: COuld you please also move it to action group ? Tests should have only Action Group according to Magento MFTF Best Practice.

**Review by edenduong — CHANGES_REQUESTED** [review]

File: app/code/Magento/Catalog/Test/Mftf/Test/AdminVerifyStorefrontPropertiesTabStillActiveAfterSaveContinueEditTest.xml (hunk 3 above)
Comment (mrtuvn) [reply app/code/Magento/Catalog/Test/Mftf/Test/AdminVerifyStorefrontPropertiesTabStillActiveAfterSaveContinueEditTest.xml]: i have updated following mftf best practice

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run WebAPI Tests, Functional Tests B2B

**Review by edenduong — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @edenduong, thank you for the review.
[ENGCOM-8976](https://jira.corp.magento.com/browse/ENGCOM-8976) has been created to process this Pull Request

**Comment by engcom-Bravo** [comment]
Hi @mrtuvn,

We are closing this PR.The PR is in Draft since longtime.

Once you are ready Please feel free to reopen it to process the PR further.

Thanks

**Pull request closed without merging**